src/index.ts

```diff
@@ -120,7 +120,7 @@ function path(template: string, params: ParamMap) {
     }
     delete remainingParams[key];
 
-    if (!allowedTypes.includes(typeof params[key])) {
+    if (!allowedTypes.includes(typeof params[key]) || typeof params[key] === "string" && params[key].trim().length <= 0) {
       throw new TypeError(
         `Path parameter ${key} cannot be of type ${typeof params[key]}. ` +
         `Allowed types are: ${allowedTypes.join(', ')}.`
```

Comment:
@balazsbotond look 👀  what to do? to make separate error message for empty string we have to check second time if params is string and its length. Is this ok?

Reply:
Yes, this should be a separate is whith another throw statement. And the error type here needs to be simply Error, not TypeError.